src/software/thunderscope/common/common_widgets.py

```diff
@@ -147,23 +148,34 @@ def setValue(self, value: float) -> None:
         """
         super(ColorProgressBar, self).setValue(int(value * self.decimals))
 
-        # clamp percent to make sure it's between 0% and 100%
-        percent = self.getPercentage()
+        def sigmoid_interpolate(val, x1, y1, x2, y2):
```

Comment:
you could use this function from C++ for this: https://github.com/UBC-Thunderbots/Software/blob/81ec0830e11d0a53002aa517a2cc190de75a9721/src/software/math/math_functions.h#L68

you would need to pybind it via `python_bindings.cpp` https://github.com/UBC-Thunderbots/Software/blob/81ec0830e11d0a53002aa517a2cc190de75a9721/src/software/python_bindings.cpp

The advantage of doing this is that the C++ one is unit tested while this one isn't  (also less code duplication)